docs/enhancements/4-nodes-cluster-deployment.md

```diff
@@ -0,0 +1,154 @@
+---
+title: 4-nodes-cluster-creation
+authors:
+  - "@flaper87"
+creation-date: 2022-04-14
+last-updated: yyyy-mm-dd
+---
+
+# Allow for a 4-nodes cluster to be created
+
+# Summary
+
+**Note:** Throughout this enhancement, a 4-nodes cluster refers to a 3 schedulable masters and 1 worker.
+
+There are cases where customers may just have 4 nodes available and no chance to add more. This was
+one of the cases in the ZTP for Factory Workflow where it was required to deploy a cluster in an
+enclosure that only had 4 nodes. The goal was to use all of the blades rather than just 3 of them.
+
+For now, this is achieved with a 2 step deployment, first 3 masters and then add a worker node. It
+would be more time efficient as well as easier and more reliable to be able to do this deployment in
+one step, day0.
+
+# Motivation
+
+Assisted service has built-in validations that prevent the deployment of a 4-nodes cluster. This is
+to prevent unhealthy clusters to be deployed. A 4-nodes cluster ends up in an unhealthy state
+because openshift-installer doesn't take schedulable masters into consideration when there are
+workers being deployed. The problem with schedulable masters not being counted as workers causes the
+installer to chose an `InfrastructureTopology` that is SingleReplica. This means that infrastructure
+workloads will run as single replicas, only on workers. This setting cannot be changed once the
+cluster is deployed.
+
+The above was brought up to the OpenShift team as [an enhancement](https://github.com/openshift/enhancements/pull/1057/). The feedback so far is that
+it's OK for users (or other installers) to overwrite the openshift manifests to achieve the
+deployment topologies required. The conversation is still in progress as far as understanding if
+some changes could be done to the openshift installer to improve the user experience here but we
+were encourage to pursue manifests' overwrite as a way to provide this through assisted service.
+
+### Goals
+
+- Allow for a 4-nodes cluster to be created using the KubeAPI flow
+
+### Non-Goals
+
+- Implement logic that will decide, for the user, what the best architecture is. Having a 3+1
+cluster will require that masters are explicitly set to schedulable and the manifests to patch the
+Infrastructure CR are provided.
+
+- Expose this through SaaS at this stage. This would require changes in the UI and other validations
+that may not worth the effort at this stage.
+
+## Proposal
+
+The biggest changes anticipated for this enhancement are:
+
+- Expose the manifests API through the InfraEnv CRD so that custom manifests can be added through
```

Comment:
A single infraenv could be used to discover hosts for many different clusters, so I think AgentClusterInstall is a better fit

Reply:
Also I think we already have such API: manifestsConfigMapRefs

Reply:
https://github.com/openshift/assisted-service/blob/e574bc2de2fd3153c7454604e32d7ee7664c1d78/api/hiveextension/v1beta1/agentclusterinstall_types.go#L120-L125